File: binary_search.py
import math
def binary_search(list,target):
    #my implementation
    # myList=list
    # myList.sort()
    # midValue=None;
    # newList=myList
    #
    # while len(newList)>0:
    #     midValue=newList[math.floor(len(newList) / 2)]
    #     if midValue==target:
    #         return myList.index(midValue)
    #     else:
    #         if midValue>target:
    #             newList = newList[0:newList.index(midValue)]
    #         else:
    #             newList= newList[newList.index(midValue)+1:]
    # return None
    first=0
    last= len(list)-1

    while first<=last:
        midpoint=(first+last)//2
        if list[midpoint]==target:
            return midpoint
        elif list[midpoint<target]:
            first=midpoint+1
        else:
            last=midpoint-1
    return None


def verifyResult(result):
    if result is not None:
        print(f"The value was found at index {result}")
    else:
        print("The value was not found on the list provided")



numbers=[1,2,3,4,5,6,7,8,9,10]
verifyResult(binary_search(numbers,10))

File: js/binary_search.js
const binarySearch = (arr, target) => {
  //   My implementation
  //   let newList = [...arr];
  //   let midpoint;
  //   while (newList.length > 0) {
  //     midpoint = newList[Math.floor(newList.length / 2)];
  //     if (midpoint === target) {
  //       let result = arr.indexOf(target);
  //       return result === 0 ? "0" : result;
  //     } else {
  //       if (midpoint > target) {
  //         newList = newList.slice(0, newList.indexOf(midpoint));
  //       } else {
  //         newList = newList.slice(newList.indexOf(midpoint + 1));
  //       }
  //     }
  //   }
  /*
   *
   *
   *
   *
   *
   * */
  //Implementation two

  let myArray = [...arr];
  let highValue = myArray.length - 1;
  let lowValue = 0;
  let midpoint;

  while (lowValue <= highValue) {
    midpoint = Math.floor((lowValue + highValue) / 2);
    // let midpointIndex = Math.floor((low + high) / 2);
    console.log("midpoint");
    console.log(midpoint);
    if (myArray[midpoint] === target) {
      console.log("found hey ", midpoint);
      return midpoint;
    } else {
      if (myArray[midpoint] > target) {
        highValue = midpoint - 1;
      } else {
        lowValue = midpoint + 1;
      }
    }
  }
};

const verifySearch = (result) => {
  result
    ? console.log(`The target was found on index ${result}`)
    : console.log("The target was not found on the provided list.");
};

myArray = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
target = 8;

verifySearch(binarySearch(myArray, target));
